java/pattern/variable/Reference.java

```diff
@@ -29,22 +29,29 @@
 
 import static com.vaticle.typedb.common.util.Objects.className;
 import static com.vaticle.typeql.lang.common.exception.ErrorMessage.INVALID_CASTING;
-import static com.vaticle.typeql.lang.common.exception.ErrorMessage.INVALID_VARIABLE_NAME;
 
 public abstract class Reference {
```

Comment:
what would this look like if we introduced `Reference.Value` and nested the rest of this under `Reference.Concept` -- then introduce `Named` as an interface?

Reply:
or something like this - to help the implementation in the server